server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs

```diff
@@ -12,66 +12,99 @@
     using Arcadia.Assistant.Calendar.Abstractions.EmployeeVacations;
     using Arcadia.Assistant.Calendar.Abstractions.EventBus;
     using Arcadia.Assistant.Calendar.Abstractions.Messages;
+    using Arcadia.Assistant.Configuration.Configuration;
     using Arcadia.Assistant.CSP.Configuration;
 
     public class CspEmployeeVacationsRegistry : UntypedActor, ILogReceive
     {
         private readonly VacationsSyncExecutor vacationsSyncExecutor;
+        private readonly IRefreshInformation refreshInformation;
         private readonly string employeeId;
 
         private readonly ILoggingAdapter logger = Context.GetLogger();
 
+        private ICancelable databaseRefreshSchedule;
+        private DatabaseVacationsCache databaseVacationsCache;
+
         public CspEmployeeVacationsRegistry(
             VacationsSyncExecutor vacationsSyncExecutor,
             AccountingReminderConfiguration reminderConfiguration,
+            IRefreshInformation refreshInformation,
             string employeeId)
         {
             this.vacationsSyncExecutor = vacationsSyncExecutor;
+            this.refreshInformation = refreshInformation;
             this.employeeId = employeeId;
 
             // Not better place to create it, but I don't know where we can do it else
             Context.ActorOf(
                 EmployeeVacationApprovedAccountingReminderActor.CreateProps(employeeId, reminderConfiguration),
                 $"vacations-reminder-{employeeId}");
 
-            this.Self.Tell(Initialization.Instance);
+            this.Self.Tell(Initialize.Instance);
         }
 
         protected override void OnReceive(object message)
         {
             switch (message)
             {
-                case Initialization _:
-                    this.GetVacations()
+                case Initialize _:
+                    this.GetVacations(false)
                         .PipeTo(
                             this.Self,
-                            success: result => new Initialization.Success(result),
-                            failure: err => new Initialization.Error(err));
+                            success: result => new Initialize.Success(result),
+                            failure: err => new Initialize.Error(err));
                     break;
 
-                case Initialization.Success msg:
-                    foreach (var @event in msg.Events)
+                case Initialize.Success msg:
+                    var eventsById = msg.Events.ToDictionary(x => x.CalendarEvent.EventId);
+                    this.databaseVacationsCache = new DatabaseVacationsCache(eventsById);
+
+                    foreach (var @event in msg.Events.Where(e => this.IsCalendarEventActual(e.CalendarEvent)))
                     {
                         Context.System.EventStream.Publish(new CalendarEventRecoverComplete(@event.CalendarEvent));
                     }
 
+                    this.ScheduleNextDatabaseRefresh();
```

Comment:
why do this instead of usual scheduling?

Reply:
Because when all events are requested I also update cache. Thereafter we need next update in 10 minutes after current moment, not after previous update.

Reply:
just send a message to self? When all events are requested you need to update it first? So you should a) send a message to update, b) forward message on get events. no?

Reply:
1) ScheduleNextDatabaseRefresh is needed to prevent getting data from the database too often. If 9 minutes have already passed since previous update and all events are requested - there is no need to update them again after 1 more minute. That's why I cancel previous scheduled updated and schedule new.

2)  Why need to send one more message, if we can just update current cache in "get events" message?

Reply:
ok.